File: integration-tests/src/test/java/dev/capylang/test/StringInterpolationTest.java
package dev.capylang.test;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class StringInterpolationTest {
    @Test
    void simpleInterpolation() {
        assertThat(StringInterpolation.simple("Capybara")).isEqualTo("Hello Capybara");
    }

    @Test
    void expressionInterpolation() {
        assertThat(StringInterpolation.expression("Capy", 7)).isEqualTo("Hello Capy7");
    }

    @Test
    void escapedInterpolationMarkersStayLiteral() {
        assertThat(StringInterpolation.escaped("ignored")).isEqualTo("Hello \\{}");
    }

    @Test
    void quoteEscapeInterpolation() {
        assertThat(StringInterpolation.quoteEscape(new StringInterpolation.Foo("Martin"))).isEqualTo("Hello \"Martin\"");
    }

    @Test
    void dataFieldInterpolation() {
        assertThat(StringInterpolation.dataField(new StringInterpolation.Foo("Martin"))).isEqualTo("Hello Martin");
    }

    @Test
    void singleQuotedStringsDoNotInterpolate() {
        assertThat(StringInterpolation.singleQuoted("Capybara")).isEqualTo("Hello {name}");
    }
}

File: capy/src/e2e-cfun/js/dev/capylang/test/StringInterpolation.test.js
const test = require('node:test');
const { assert, generatedModule } = require('./_support');

test('StringInterpolation', () => {
    const interpolation = generatedModule('dev/capylang/test/StringInterpolation.js');
    assert.equal(interpolation.simple('Capybara'), 'Hello Capybara');
    assert.equal(interpolation.expression('Capy', 7), 'Hello Capy7');
    assert.equal(interpolation.escaped('ignored'), 'Hello \\{}');
    assert.equal(interpolation.escapedWithInterpolationValue('boo'), 'xyz {boo}');
    assert.equal(interpolation.quoteEscape(new interpolation.Foo({ name: 'Martin' })), 'Hello "Martin"');
    assert.equal(interpolation.dataField(new interpolation.Foo({ name: 'Martin' })), 'Hello Martin');
    assert.equal(interpolation.singleQuoted('Capybara'), 'Hello {name}');
});
